### PR TITLE
Add linkerd support, bump chart and supporting charts versions

### PR DESCRIPTION
**Pull request by dusandordevicsap**
This adds the linkerd-support chart as dependency and also adds the
annotations to all Deployment, Service, Daemonset and Ingress objects.

linkerd is disabled by default and needs to be enabled per region.

We depend on the utils chart for the annotation snippets and the
conditions so our templates stay more compact and thus better readable.

Supporting charts versions are bumped to support this change.

Chart version bumped to 0.3.11

### CODE REVIEW

File: openstack/manila/Chart.yaml [hunk 1]
```diff
@@ -9,16 +9,16 @@ maintainers:
...
   - condition: mariadb.enabled
     name: mariadb
     repository: https://charts.eu-de-2.cloud.sap
-    version: ~0.7.6
+    version: 0.8.0
   - condition: memcached.enabled
     name: memcached
     repository: https://charts.eu-de-2.cloud.sap
-    version: ~0.1.0
+    version: 0.1.5
```
Comment (joker-at-work) [inline openstack/manila/Chart.yaml:21]: This needs to be `2.0.0` at least.

File: openstack/manila/templates/api-service.yaml [hunk 2]
```diff
@@ -10,7 +10,9 @@ metadata:
   annotations:
     prometheus.io/scrape: "true"
     prometheus.io/targets: {{ required ".Values.alerts.prometheus.openstack" .Values.alerts.prometheus.openstack |  quote }}
+    {{- include "utils.linkerd.pod_and_service_annotation" . | indent 4 }}
   {{- include "utils.topology.service_topology_mode" . | indent 2 }}
+
```
Comment (joker-at-work) [inline openstack/manila/templates/api-service.yaml:15]: Did you add this newline on purpose?

**Review by joker-at-work — CHANGES_REQUESTED** [review]
It looks like you missed 2 deployments:
```
templates/shares/_netapp-deployment.yaml.tpl:kind: Deployment
templates/shares/_netapp-ensure-deployment.yaml.tpl:kind: Deployment
```

File: openstack/manila/templates/api-service.yaml (hunk 2 above)
Comment (dusandordevicsap) [reply openstack/manila/templates/api-service.yaml]: May be typo, I am not sure, removed it

**Review by joker-at-work — APPROVED** [review]
I cannot see any obvious problems, but also haven't rendered the chart.

**Pull request merged**